### PR TITLE
Refactor ci build

### PR DESCRIPTION
**Pull request by zaro0508**
* Use Gemfile to manage fastlane dependencies
* move call to tests into build.sh
* don't need to install fastlane, it's already on travis.
  gem update handles updating fastlane.
* build docs only when tag is created
* fix deployment of docs to s3 bucket.
* turn off testing for PRs, travis does not support secrets in PRs

### CODE REVIEW

File: travis/build.sh [hunk 1]
```diff
@@ -3,11 +3,12 @@ set -ex
...
 if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
-    bundle exec fastlane test scheme:"BridgeSDK"
-elif [ "$TRAVIS_BRANCH" = "master" ]; then
-    bundle exec fastlane test scheme:"BridgeSDK"
+    bundle exec fastlane doc scheme:"BridgeSDK"
 fi
-if [ ! -z "$TRAVIS_TAG" ]; then
+if [ "$TRAVIS_PULL_REQUEST" = "false" ]; then
     bundle exec fastlane doc scheme:"BridgeSDK"
+    openssl aes-256-cbc -K $encrypted_d59c41fdd72b_key -iv $encrypted_d59c41fdd72b_iv -in BridgeAdminCredentials.plist.enc -out BridgeAdminCredentials.plist -d
```
Comment (zaro0508) [inline travis/build.sh]: We've decided setup ENV vars for secrets instead of committing encrypted files to the repo.

File: .travis.yml [hunk 2]
```diff
@@ -8,6 +8,9 @@ before_install:
 - brew update
 - bundle update
 - echo -e "machine github.com\n  login $CI_USER_TOKEN" >> ~/.netrc
+- openssl aes-256-cbc -K $encrypted_d59c41fdd72b_key -iv $encrypted_d59c41fdd72b_iv
```
Comment (zaro0508) [inline .travis.yml]: We've decided to use ENV vars for secrets instead of committing encrypted files to the repo

File: travis/build.sh (hunk 1 above)
Comment (zaro0508) [inline travis/build.sh]: We've decided to use ENV vars for secrets instead of committing encrypted files to the repo

File: fastlane/Fastfile [hunk 3]
```diff
@@ -18,16 +18,17 @@ default_platform :ios
...
     
   end
 
   desc "Build the documentation"
   lane :doc do |options|
+    scheme = options[:scheme]
     appledoc(
-      project_name: "BridgeSDK",
+      project_name: "#{scheme}",
       project_company: "Sage BioNetworks",
```
Comment (Erin-Mounts) [inline fastlane/Fastfile]: should be lowercase 'n' in Bionetworks

**Pull request merged**